core/unix/signal_linux_aarch64.c

```diff
@@ -54,21 +54,123 @@ save_fpstate(dcontext_t *dcontext, sigframe_rt_t *frame)
 }
 
 #ifdef DEBUG
+/* Representation of quadwords as 2 doublewords. */
+typedef union {
+    __uint128_t as_128;
+    struct {
+        uint64 lo;
+        uint64 hi;
+    } as_2x64;
+} reinterpret128_2x64_t;
+
 void
 dump_sigcontext(dcontext_t *dcontext, sigcontext_t *sc)
 {
+#    ifdef DR_HOST_NOT_TARGET
+    ASSERT_NOT_REACHED();
+#    endif
+    LOG(THREAD, LOG_ASYNCH, 1, "\tSignal context:\n");
     int i;
     for (i = 0; i <= DR_REG_X30 - DR_REG_X0; i++)
         LOG(THREAD, LOG_ASYNCH, 1, "\tx%-2d    = " PFX "\n", i, sc->regs[i]);
     LOG(THREAD, LOG_ASYNCH, 1, "\tsp     = " PFX "\n", sc->sp);
     LOG(THREAD, LOG_ASYNCH, 1, "\tpc     = " PFX "\n", sc->pc);
     LOG(THREAD, LOG_ASYNCH, 1, "\tpstate = " PFX "\n", sc->pstate);
+    LOG(THREAD, LOG_ASYNCH, 1, "\n");
+
+    struct _aarch64_ctx *head = (struct _aarch64_ctx *)sc->__reserved;
+    ASSERT(head->magic == FPSIMD_MAGIC);
+    ASSERT(head->size == sizeof(struct fpsimd_context));
+
+    struct fpsimd_context *fpsimd = (struct fpsimd_context *)sc->__reserved;
+    LOG(THREAD, LOG_ASYNCH, 2, "\tfpsr 0x%x\n", fpsimd->fpsr);
+    LOG(THREAD, LOG_ASYNCH, 2, "\tfpcr 0x%x\n", fpsimd->fpcr);
+    reinterpret128_2x64_t vreg;
+    for (i = 0; i < MCXT_NUM_SIMD_SVE_SLOTS; i++) {
+        vreg.as_128 = fpsimd->vregs[i];
+        LOG(THREAD, LOG_ASYNCH, 2, "\tq%-2d  0x%016lx %016lx\n", i, vreg.as_2x64.hi,
+            vreg.as_2x64.lo);
+    }
+    LOG(THREAD, LOG_ASYNCH, 2, "\n");
+
+    if (proc_has_feature(FEATURE_SVE)) {
+        size_t offset = sizeof(struct fpsimd_context);
+        struct _aarch64_ctx *next_head = (struct _aarch64_ctx *)(sc->__reserved + offset);
+        while (next_head->magic != 0) {
+            switch (next_head->magic) {
+            case ESR_MAGIC: break;
+            case EXTRA_MAGIC: break;
+            case SVE_MAGIC: {
+                const struct sve_context *sve = (struct sve_context *)(next_head);
+                LOG(THREAD, LOG_ASYNCH, 2, "\tSVE VL %d bytes\n", sve->vl);
+                ASSERT(sve->vl == proc_get_vector_length_bytes());
+                const unsigned int vq = sve_vq_from_vl(sve->vl);
```

Comment:
Echoing comments above: is there a more descriptive name?  I don't know what "vq" is; "sve_vq_from_vl" is similarly hard to understand.

Reply:
Changed `vq` to `quads_per_vector` and `sve_vq_from_vl()` to `sve_vecquad_from_veclen()`.
Added comment describing rationale for using quadwords.